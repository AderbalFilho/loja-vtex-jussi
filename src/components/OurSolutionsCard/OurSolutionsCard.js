import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import { JButton, JCard, JText } from '@ui';

import './OurSolutionsCard.scss';

function OurSolutionsCard({ classes, solution }) {
  const className = classNames('our-solutions-card', classes);

  return (
    <JCard classes={className}>
      <img
        src={solution.img}
        alt={solution.imgAlt}
        className="our-solutions-card__img"
      />
      <div>
        <JText color="black" classes="our-solutions-card__name">
          {solution.name}
        </JText>
        <JText
          size="small"
          color="pink"
          classes="our-solutions-card__description"
        >
          {solution.description}
        </JText>
        {solution.features && (
          <ul className="our-solutions-card__list">
            {solution.features.map((feature) => {
              return (
                <li key={feature}>
                  <JText color="dark-grey" size="small">
                    {feature}
                  </JText>
                </li>
              );
            })}
          </ul>
        )}
        <JButton
          tag="a"
          href={solution.link}
          rel="noreferrer noopener"
          target="_blank"
          color="black"
          border="green"
          backgroundColor="green"
          classes="our-solutions-card__button"
        >
          Ver solução
        </JButton>
      </div>
    </JCard>
  );
}

OurSolutionsCard.propTypes = {
  classes: PropTypes.string,
  solution: PropTypes.shape({
    description: PropTypes.string.isRequired,
    features: PropTypes.arrayOf(PropTypes.string).isRequired,
    img: PropTypes.string.isRequired,
    imgAlt: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default OurSolutionsCard;
