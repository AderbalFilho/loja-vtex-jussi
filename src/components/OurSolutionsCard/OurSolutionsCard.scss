.our-solutions-card {
  padding: 16px;
}

.our-solutions-card__img {
  border-radius: 50%;
  height: 100px;
  margin-bottom: 40px;
  width: 100px;
}

.our-solutions-card__name {
  margin: 0;
}

.our-solutions-card__description {
  margin: 16px 0;
}

.our-solutions-card__list {
  list-style: none;
  margin: 0 0 16px 0;

  p {
    display: inline-block;
    margin: 0 0 1px 0;
  }

  li::before {
    display: inline-block;
    font-weight: bold;
    color: var(--dark-grey);
    content: '\2022';
    margin-left: -40px;
    width: 0.5rem;
  }
}

.our-solutions-card__button {
  display: block;
  text-align: center;
}
