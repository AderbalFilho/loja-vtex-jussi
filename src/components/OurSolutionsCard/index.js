import OurSolutionsCard from './OurSolutionsCard';

export default OurSolutionsCard;
