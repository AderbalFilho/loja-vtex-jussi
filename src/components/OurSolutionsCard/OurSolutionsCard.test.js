import React from 'react';
import TestRenderer from 'react-test-renderer';

import OurSolutionsCard from './OurSolutionsCard';

describe('OurSolutionsCard', () => {
  const solution = {
    description: 'Descrição do produto #1',
    features: ['Feature 1', 'Feature 2', 'Feature 3'],
    img:
      'https://static.openfoodfacts.org/images/products/073/762/806/4502/ingredients_en.10.400.jpg',
    imgAlt: 'P1',
    link: '',
    name: 'Nome do Produto #1',
  };

  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(
      <OurSolutionsCard solution={solution} />
    );
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });
});
