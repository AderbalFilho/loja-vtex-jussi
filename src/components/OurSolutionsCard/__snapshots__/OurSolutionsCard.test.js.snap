// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OurSolutionsCard matches snapshot 1`] = `
<div
  className="j-card j-card--rad-normal j-card--size-normal our-solutions-card"
>
  <img
    alt="P1"
    className="our-solutions-card__img"
    src="https://static.openfoodfacts.org/images/products/073/762/806/4502/ingredients_en.10.400.jpg"
  />
  <div>
    <p
      className="j-text j-text--black j-text--normal our-solutions-card__name"
    >
      Nome do Produto #1
    </p>
    <p
      className="j-text j-text--pink j-text--small our-solutions-card__description"
    >
      Descrição do produto #1
    </p>
    <ul
      className="our-solutions-card__list"
    >
      <li>
        <p
          className="j-text j-text--dark-grey j-text--small"
        >
          Feature 1
        </p>
      </li>
      <li>
        <p
          className="j-text j-text--dark-grey j-text--small"
        >
          Feature 2
        </p>
      </li>
      <li>
        <p
          className="j-text j-text--dark-grey j-text--small"
        >
          Feature 3
        </p>
      </li>
    </ul>
    <a
      className="j-button j-button--back-green j-button--border-green j-button--black j-button--rad-normal j-button--link our-solutions-card__button"
      href=""
      rel="noreferrer noopener"
      target="_blank"
    >
      Ver solução
    </a>
  </div>
</div>
`;
